lotus/metrics.go

```diff
@@ -13,4 +19,123 @@ var (
 		Description: "Current height of Lotus node",
 		Aggregation: view.LastValue(),
 	}
+
+	metricHeightInterval = time.Second * 120
 )
+
+// SyncMonitor provides information about the Lotus
+// syncing status.
+type SyncMonitor struct {
+	cb ClientBuilder
+
+	lock       sync.Mutex
+	heightDiff int64
+	remaining  int64
+}
+
+// NewSyncMonitor creates a new LotusSyncMonitor.
+func NewSyncMonitor(cb ClientBuilder) (*SyncMonitor, error) {
+	if err := view.Register(vHeight); err != nil {
+		log.Fatalf("register metrics views: %v", err)
+	}
+	lsm := &SyncMonitor{cb: cb}
+
+	if err := lsm.refreshSyncDiff(); err != nil {
+		return nil, fmt.Errorf("getting initial sync height diff: %s", err)
+	}
+
+	go func() {
+		for {
+			lsm.evaluate()
+			time.Sleep(metricHeightInterval)
+		}
+	}()
+	return lsm, nil
+}
+
+// SyncHeightDiff returns the height difference between the tip of the chain
+// and the current synced height.
+func (lsm *SyncMonitor) SyncHeightDiff() int64 {
+	lsm.lock.Lock()
+	defer lsm.lock.Unlock()
+	return lsm.heightDiff
+}
```

Comment:
Provide cached value... we don't want people asking lotus about its sync status to add load to it.
Since the syncing process is rather slow, a value 2 minutes old is usually good enough to make decisions.